<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1050</width>
    <height>857</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string notr="true" extracomment="Serial console tool for programmers.">QIoT-terminal</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_13">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
       <widget class="QFrame" name="frame">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_11">
         <item>
          <widget class="QPushButton" name="pushButton_connectToggle">
           <property name="text">
            <string>Connect</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_about">
           <property name="text">
            <string>About</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_Port_info_toggle">
           <property name="text">
            <string>Port info</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="title">
         <string>Port settings</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Port</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="comboBox_ports"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Baud rate</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="comboBox_baudrates"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <item>
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Custom baudrate</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineEdit_baudrate"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_7">
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Data bits</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="comboBox_databits"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_10">
           <item>
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Parity</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="comboBox_parity"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_8">
           <item>
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Stop bits</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="comboBox_stopbits"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_14">
           <item>
            <widget class="QLabel" name="label_12">
             <property name="text">
              <string>FlowControl</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="comboBox_flowcontrol"/>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>85</height>
       </size>
      </property>
      <property name="title">
       <string>Receive</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QPushButton" name="pushButton_CLEAR">
         <property name="text">
          <string>CLEAR</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QRadioButton" name="radioButton_HEX">
           <property name="text">
            <string>HEX</string>
           </property>
           <attribute name="buttonGroup">
            <string notr="true">buttonGroup</string>
           </attribute>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radioButton_ASCII">
           <property name="text">
            <string>ASCII</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
           <attribute name="buttonGroup">
            <string notr="true">buttonGroup</string>
           </attribute>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QCheckBox" name="checkBox_Dec">
           <property name="text">
            <string>Dec</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="checkBox_Hex">
           <property name="text">
            <string>Hex</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QCheckBox" name="checkBox_Bin">
         <property name="text">
          <string>Bin</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="sizeConstraint">
       <enum>QLayout::SetNoConstraint</enum>
      </property>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QPlainTextEdit" name="plainTextEdit_port_info_window">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="font">
           <font>
            <family>Ubuntu Mono</family>
            <pointsize>9</pointsize>
           </font>
          </property>
          <property name="cursor" stdset="0">
           <cursorShape>IBeamCursor</cursorShape>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="plainText">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="plainTextEdit_console">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="font">
           <font>
            <family>Ubuntu Mono</family>
           </font>
          </property>
          <property name="cursor" stdset="0">
           <cursorShape>IBeamCursor</cursorShape>
          </property>
          <property name="verticalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOn</enum>
          </property>
          <property name="horizontalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOff</enum>
          </property>
          <property name="undoRedoEnabled">
           <bool>false</bool>
          </property>
          <property name="lineWrapMode">
           <enum>QPlainTextEdit::WidgetWidth</enum>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="plainText">
           <string notr="true"/>
          </property>
          <property name="backgroundVisible">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="plainTextEdit_HEX">
          <property name="minimumSize">
           <size>
            <width>65</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>55</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="font">
           <font>
            <family>Ubuntu Mono</family>
           </font>
          </property>
          <property name="cursor" stdset="0">
           <cursorShape>IBeamCursor</cursorShape>
          </property>
          <property name="verticalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOn</enum>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="plainText">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="plainTextEdit_DEC">
          <property name="minimumSize">
           <size>
            <width>65</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>55</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="font">
           <font>
            <family>Ubuntu Mono</family>
           </font>
          </property>
          <property name="cursor" stdset="0">
           <cursorShape>IBeamCursor</cursorShape>
          </property>
          <property name="verticalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOn</enum>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="plainText">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="plainTextEdit_BIN">
          <property name="minimumSize">
           <size>
            <width>130</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>55</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="font">
           <font>
            <family>Ubuntu Mono</family>
            <kerning>true</kerning>
           </font>
          </property>
          <property name="cursor" stdset="0">
           <cursorShape>IBeamCursor</cursorShape>
          </property>
          <property name="verticalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOn</enum>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="plainText">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>117</height>
       </size>
      </property>
      <property name="title">
       <string>Transmit</string>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
      <property name="checkable">
       <bool>false</bool>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_12">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QLineEdit" name="lineEdit_send_input">
           <property name="font">
            <font>
             <family>Serif</family>
            </font>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_send">
           <property name="text">
            <string>Send</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPlainTextEdit" name="plainTextEdit">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>45</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>45</height>
          </size>
         </property>
         <property name="font">
          <font>
           <family>Serif</family>
          </font>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
         <property name="plainText">
          <string/>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_9">
      <item>
       <widget class="QCheckBox" name="checkBox_appendLinefeed">
        <property name="text">
         <string>Append NL line feed (0x0A)</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="checkBox_appendCarriageReturn">
        <property name="text">
         <string>Append CR carriage return (0x0D)</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton_Port_info_toggle</sender>
   <signal>toggled(bool)</signal>
   <receiver>plainTextEdit_port_info_window</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>57</x>
     <y>83</y>
    </hint>
    <hint type="destinationlabel">
     <x>106</x>
     <y>204</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_Dec</sender>
   <signal>toggled(bool)</signal>
   <receiver>plainTextEdit_DEC</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>831</x>
     <y>149</y>
    </hint>
    <hint type="destinationlabel">
     <x>762</x>
     <y>203</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_Hex</sender>
   <signal>toggled(bool)</signal>
   <receiver>plainTextEdit_HEX</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>831</x>
     <y>175</y>
    </hint>
    <hint type="destinationlabel">
     <x>703</x>
     <y>207</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_Bin</sender>
   <signal>toggled(bool)</signal>
   <receiver>plainTextEdit_BIN</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>856</x>
     <y>159</y>
    </hint>
    <hint type="destinationlabel">
     <x>833</x>
     <y>205</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <buttongroups>
  <buttongroup name="buttonGroup"/>
 </buttongroups>
</ui>
